src/snarks/groth16/evaluation_from_lagrange.tcc

```diff
@@ -0,0 +1,37 @@
+#pragma once
```

Comment:
I'd stick to:
```cpp
#ifndef __FILE_EXTENSION__
#define __FILE_EXTENSION__ 
...
#endif
```

This is applicable to several files in this PR that also use `#pragma once` as an alternative

Reply:
Then I'd include the `.tcc` file at the end of the `.hpp` file